api/src/main/java/jakarta/enterprise/inject/spi/BeanContainer.java

```diff
@@ -278,21 +281,24 @@ boolean isMatchingBean(Set<Type> beanTypes, Set<Annotation> beanQualifiers, Type
             Set<Annotation> requiredQualifiers);
 
     /**
-     * Returns {@code true} if an event object with given type and qualifiers would match
+     * Returns {@code true} if an event object with given runtime type and qualifiers would match
      * an observer method with given observed event type and observed event qualifiers, {@code false} otherwise.
      * <p>
      * In line with the specification for events and observer resolution, the set of
-     * {@code eventQualifiers} is considered to always include {@code @Any}.
+     * {@code eventQualifiers} is considered to always include {@code @Any} and also include
+     * {@code @Default} when it contains no other qualifier but {@code @Any}.
```

Comment:
The spec does not say that an event has the `@Default` qualifier if it has no qualifiers other than `@Any`.

It instead specifies that different behaviour is applied to _observers_ with `@Default` qualifier.

> Such observer will only be notified for events having either no qualifiers or only @Default qualifier:

Although I think both ways of specifying this have the same result, we should not have the Javadoc here contradicting the spec.

Reply:
Yeah, you're right. I think I found a way to rewrite the text, but it required me to change the terms "event type" and "event qualifiers" to "specified type" and "specified qualifiers" (which are also CDI specification terms, I'm not inventing anything new here).